server/src/test/java/org/elasticsearch/action/admin/indices/template/get/GetComposableIndexTemplateResponseTests.java

```diff
@@ -26,19 +28,24 @@ protected Writeable.Reader<GetComposableIndexTemplateAction.Response> instanceRe
 
     @Override
     protected GetComposableIndexTemplateAction.Response createTestInstance() {
-        DataStreamGlobalRetention globalRetention = randomBoolean() ? null : DataStreamGlobalRetentionTests.randomGlobalRetention();
+        RolloverConfiguration rolloverConfiguration = randomBoolean() ? null : RolloverConfigurationTests.randomRolloverConditions();
         if (randomBoolean()) {
-            return new GetComposableIndexTemplateAction.Response(Map.of(), globalRetention);
+            return new GetComposableIndexTemplateAction.Response(Map.of(), rolloverConfiguration);
         }
         Map<String, ComposableIndexTemplate> templates = new HashMap<>();
         for (int i = 0; i < randomIntBetween(1, 4); i++) {
             templates.put(randomAlphaOfLength(4), ComposableIndexTemplateTests.randomInstance());
         }
-        return new GetComposableIndexTemplateAction.Response(templates, globalRetention);
+        return new GetComposableIndexTemplateAction.Response(templates, rolloverConfiguration);
     }
 
     @Override
     protected GetComposableIndexTemplateAction.Response mutateInstance(GetComposableIndexTemplateAction.Response instance) {
         return randomValueOtherThan(instance, this::createTestInstance);
```

Comment:
Just a note -- this kind of mutateInstance seems to defeat the purpose of the test, but that's outside of the scope of this PR.

Reply:
Happy to implement since I am in the code :)